kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/assertions/nondeterministic/continually.kt

```diff
@@ -122,3 +128,31 @@ private fun <T> ContinuallyConfigurationBuilder<T>.build(): ContinuallyConfigura
       },
    )
 }
+
+
+/**
+ * Store the [TimeSource] used by [continually].
+ *
+ * @see getContinuallyTimeSource
+ */
+internal class ContinuallyTimeSource(
+   val timeSource: TimeSource
+) : CoroutineContext.Element {
+
+   override val key: CoroutineContext.Key<ContinuallyTimeSource>
+      get() = KEY
+
+   internal companion object {
+      /**
+       * Retrieves the [TimeSource] used by [continually].
+       *
+       * For internal Kotest testing purposes the [TimeSource] can be overridden.
+       * For normal usage [TimeSource.Monotonic] is used.
+       */
+      internal suspend fun getContinuallyTimeSource(): TimeSource =
+         coroutineContext[KEY]?.timeSource
+            ?: TimeSource.Monotonic
+
+      internal val KEY = object : CoroutineContext.Key<ContinuallyTimeSource> {}
+   }
+}
```

Comment:
In the 'retry' tests PR https://github.com/kotest/kotest/pull/4101 I added a `val timeSource` to the configuration. Here I'm trying a different approach and storing the TimeSource in the CoroutineContext. I think this is better, because it makes it easier to test any `continually()` functions, and it doesn't expose any options for users (even the option is guarded with `@KotestInternal`). I'd like to revisit the `retry` tests, and refactor the `eventually` tests, to do the same.